atc/db/team.go

```diff
@@ -1258,7 +1246,7 @@ func saveResource(tx Tx, resource atc.ResourceConfig, pipelineID int) (int, erro
 	err = psql.Insert("resources").
 		Columns("name", "pipeline_id", "config", "active", "nonce", "type").
 		Values(resource.Name, pipelineID, encryptedPayload, true, nonce, resource.Type).
-		Suffix("ON CONFLICT (name, pipeline_id) DO UPDATE SET config = EXCLUDED.config, active = EXCLUDED.active, nonce = EXCLUDED.nonce, type = EXCLUDED.type").
+		Suffix("ON CONFLICT (name, pipeline_id) DO UPDATE SET config = EXCLUDED.config, active = EXCLUDED.active, nonce = EXCLUDED.nonce, type = EXCLUDED.type, resource_config_id = NULL, resource_config_scope_id = NULL").
```

Comment:
I stand corrected here that @clarafu just pointed out that the implementation here will set ALL existing resources' resource_config_id to null. In this case, there will be a window of resource config collector to GC those un-linked resource configs and causing all existing versions to be deleted.

https://github.com/concourse/concourse/blob/d8d27b9bd997db0e5bce5ed4f3c8dbe9b1b43466/atc/db/resource_config_factory.go#L236-L243

Note the delete cascade chain is resource_config_versions(resource_config_scopes.id) -> resource_config_scopes(resource_configs.id)->resource_configs